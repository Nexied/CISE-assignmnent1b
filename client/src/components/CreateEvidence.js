/* eslint-disable no-console */
/* eslint-disable no-undef */
/* eslint-disable react/no-unused-state */
/* eslint-disable no-unused-vars */
import React, { Component } from "react"
import { Link } from "react-router-dom"
import axios from "axios"

// this is will be use in the near future to submit an evidence
class CreateEvidence extends Component {
  constructor() {
    super()
    this.state = {
      article: "",
      author: "",
      title: "",
      journal: "",
      eprint: "",
      eprinttype: "",
      eprintclass: "",
      pages: "",
      month: "",
      annote: "",
      claimStrength: ""
    }
  }
}

onChange = e => {
  this.setState({ [e.target.name]: e.target.value })
}

onSubmit = e => {
  e.preventDefault()

  const data = {
    article: this.state.article,
    author: this.state.author,
    title: this.state.title,
    journal: this.state.journal,
    eprint: this.state.eprint,
    eprinttype: this.state.eprinttype,
    eprintclass: this.state.eprintclass,
    pages: this.state.pages,
    month: this.state.month,
    annote: this.state.annote,
    claimStrength: this.state.claimStrength
  }

  axios
    .post("http://localhost:8082/api/evidences", data)
    .then(res => {
      this.setState({
        article: "",
        author: "",
        title: "",
        journal: "",
        eprint: "",
        eprinttype: "",
        eprintclass: "",
        pages: "",
        month: "",
        annot: "",
        claimStrength: ""
      })
      this.props.history.push("/")
    })
    .catch(err => {
      console.log("error in create evidence")
    })
}
