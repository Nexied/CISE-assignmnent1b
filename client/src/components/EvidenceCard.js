/* eslint-disable no-console */
/* eslint-disable no-restricted-exports */
/* eslint-disable import/prefer-default-export */
/* eslint-disable react/prop-types */
import React from "react"

export const EvidenceCard = ({ evidenceData }) => {
  const evidence = evidenceData
  console.log(evidence.article)

  return (
    <tr>
      <td>{evidence.title}</td>
      <td>{evidence.author}</td>
      <td>{evidence.year}</td>
      <td>{evidence.sePractice}</td>
      <td>{evidence.claim}</td>
      <td>{evidence.claimStrength}</td>
    </tr>
  )
}
