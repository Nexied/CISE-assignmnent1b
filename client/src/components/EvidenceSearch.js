/* eslint-disable no-console */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable import/prefer-default-export */
/* eslint-disable no-unused-vars */
/* eslint-disable import/no-extraneous-dependencies */
import React, { useState, useEffect } from "react"
import Select from "react-select"
import { Link } from "react-router-dom"
import { EvidenceCard } from "./EvidenceCard"

export const EvidenceSearch = () => {
  const [seMethods, setSEMethods] = useState([])
  const [claims, setClaims] = useState([])
  const [searchInfo, setSearchInfo] = useState({
    seMethod: "",
    claim: "",
    startYear: "",
    endYear: ""
  })

  const [showResults, setShowResults] = useState(false)
  const [evidenceCardTest, setEvidenceCards] = useState([])

  // Temporary SE Methods list
  const seList = [
    { label: "TDD", value: "TDD" },
    { label: "FDD", value: "FDD" },
    { label: "SCRUM", value: "SCRUM" },
    { label: "UNIT-TESTING", value: "UNIT-TESTING" },
    { label: "AGILE", value: "AGILE" }
  ]

  // TODO: Try get the list of SE methods from a database/script list
  const claimsList = [
    { label: "Improves Code Quality", value: "Improves Code Quality" },
    {
      label: "More efficient code production",
      value: "Code Production Efficiency"
    },
    { label: "Improve Team Quality", value: "Improve team quality" },
    { label: "Improve App Quality", value: "Improve app quality" }
  ]

  useEffect(() => {
    console.log("Website loaded!")
    // console.log("Print id: " + props.match.params.id);
  }, [])

  const onChange = e => {
    console.log(e)
    setSearchInfo({ ...searchInfo, [e.target.name]: e.target.value })
    console.log(searchInfo)
  }

  const onSelectChange = (e, attribute) => {
    console.log(e.value)
    console.log(`The actual attribute: ${attribute}`)
    setSearchInfo({ ...searchInfo, [attribute]: e.value })
  }

  const onSubmit = e => {
    e.preventDefault()
  }

  return (
    <div
      className="container-fluid bg-dark text-light"
      style={{ height: "100vh" }}
    >
      <div className="row">
        <div className="col-12">{/* Nav bar here? */}</div>
      </div>
      <div className="row">
        <div className="col-12">
          <h1 className="display-2">SEEDS Version 1</h1>
        </div>
      </div>
      <div className="row align-items-center">
        <div className="col-12" align="center">
          <form>
            <div className="form-group col-3">
              <label className="lead fs-2">SE Practice:</label>
              <Select
                className="text-dark"
                onChange={e => {
                  onSelectChange(e, "seMethod")
                }}
                options={seList}
              />
              <br />

              <label className="lead fs-2">Claim:</label>
              <Select
                className="text-dark"
                onChange={e => onSelectChange(e, "claim")}
                options={claimsList}
              />
              <br />

              <label className="lead fs-2">Year Range:</label>
              <input
                className="form-control"
                type="text"
                name="startYear"
                onChange={onChange}
                maxLength="4"
                placeholder="Start Year e.g. 2000"
              />
              <input
                className="form-control"
                type="text"
                name="endYear"
                onChange={onChange}
                maxLength="4"
                placeholder="End Year e.g. 2010"
              />

              <button
                className="btn btn-default bg-dark text-light border border-light mt-3"
                type="submit"
                onSubmit={onSubmit}
              >
                Search
              </button>

              <Link
                to={{
                  pathname: `/show-results`,
                  searchParams: { ...searchInfo }
                }}
                className="btn btn-outline-info btn-lg btn-block"
              >
                View Results
              </Link>
            </div>
          </form>
        </div>
        <div>{evidenceCardTest}</div>
      </div>
    </div>
  )
}
