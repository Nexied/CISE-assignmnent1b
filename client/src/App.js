/* eslint-disable import/order */
/* eslint-disable react/react-in-jsx-scope */
import { EvidenceSearch } from "./components/EvidenceSearch"
import { ShowEvidenceResults } from "./components/ShowEvidenceResults"
import { BrowserRouter as Router, Route } from "react-router-dom"

function App() {
  return (
    <Router>
      <div>
        <Route exact path="/" component={EvidenceSearch} />
        <Route path="/show-results" component={ShowEvidenceResults} />
      </div>
    </Router>
  )
}

export default App
