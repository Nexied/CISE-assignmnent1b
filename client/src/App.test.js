import { render } from "@testing-library/react"
import React from "react";
import { App } from "./App";
import { EvidenceSearch } from "./components/EvidenceSearch"
import { ShowEvidenceResults } from "./components/ShowEvidenceResults"
import { BrowserRouter as Router, Route } from "react-router-dom"


test("render it", () => {
    const component = render(
        <Router>
            <div>
                <Route exact path="/" component={EvidenceSearch} />
                <Route path="/show-results" component={ShowEvidenceResults} />
            </div>
        </Router>
    )
    expect(component.container).toMatchSnapshot()
})

