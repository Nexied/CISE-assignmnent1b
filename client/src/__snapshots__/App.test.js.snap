// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render it 1`] = `
<div>
  <div>
    <div
      class="container-fluid bg-dark text-light"
      style="height: 100vh;"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h1
            class="display-2"
          >
            SEEDS Version 1
          </h1>
        </div>
      </div>
      <div
        class="row align-items-center"
      >
        <div
          align="center"
          class="col-12"
        >
          <form>
            <div
              class="form-group col-3"
            >
              <label
                class="lead fs-2"
              >
                SE Practice:
              </label>
              <div
                class="text-dark css-2b097c-container"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-yk16xz-control"
                >
                  <div
                    class=" css-g1d714-ValueContainer"
                  >
                    <div
                      class=" css-1wa3eu0-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class=""
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-2-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <br />
              <label
                class="lead fs-2"
              >
                Claim:
              </label>
              <div
                class="text-dark css-2b097c-container"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-yk16xz-control"
                >
                  <div
                    class=" css-g1d714-ValueContainer"
                  >
                    <div
                      class=" css-1wa3eu0-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class=""
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-3-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <br />
              <label
                class="lead fs-2"
              >
                Year Range:
              </label>
              <input
                class="form-control"
                maxlength="4"
                name="startYear"
                placeholder="Start Year e.g. 2000"
                type="text"
              />
              <input
                class="form-control"
                maxlength="4"
                name="endYear"
                placeholder="End Year e.g. 2010"
                type="text"
              />
              <button
                class="btn btn-default bg-dark text-light border border-light mt-3"
                type="submit"
              >
                Search
              </button>
              <a
                class="btn btn-outline-info btn-lg btn-block"
                href="/show-results"
              >
                View Results
              </a>
            </div>
          </form>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
